src/Dialog/Dialog.js

```diff
@@ -56,100 +162,11 @@ export const styleSheet = createStyleSheet('MuiDialog', (theme) => {
  * </Dialog>
  * ```
  */
-export default class Dialog extends Component {
-
-  static propTypes = {
-    /**
-     * Dialog children, usually the included sub-components.
-     */
-    children: PropTypes.node,
-    /**
-     * The CSS class name of the root element.
-     */
-    className: PropTypes.string,
-    /**
-     * If `true`, the dialog will be full-screen.
-     */
-    fullScreen: PropTypes.bool,
-    /**
-     * If `true`, clicking the backdrop will not fire the `onRequestClose` callback.
-     */
-    ignoreBackdropClick: PropTypes.bool,
-    /**
-     * If `true`, hitting escape will not fire the `onRequestClose` callback.
-     */
-    ignoreEscapeKeyUp: PropTypes.bool,
-    /**
-     * Duration of the animation when the element is entering.
-     */
-    enterTransitionDuration: PropTypes.number, // eslint-disable-line react/sort-prop-types
-    /**
-     * Duration of the animation when the element is leaving.
-     */
-    leaveTransitionDuration: PropTypes.number,
-    /**
-     * Determine the max width of the dialog.
-     * The dialog width grows with the size of the screen, this property is useful
-     * on the desktop where you might need some coherent different width size across your
-     * application.
-     */
-    maxWidth: PropTypes.oneOf([
-      'xs',
-      'sm',
-      'md',
-    ]),
-    /**
-     * Callback fired when the backdrop is clicked.
-     */
-    onBackdropClick: PropTypes.func,
-    /**
-     * Callback fired before the dialog enters.
-     */
-    onEnter: PropTypes.func,
-    /**
-     * Callback fired when the dialog is entering.
-     */
-    onEntering: PropTypes.func,
-    /**
-     * Callback fired when the dialog has entered.
-     */
-    onEntered: PropTypes.func, // eslint-disable-line react/sort-prop-types
-    /**
-     * Callback fires when the escape key is released and the modal is in focus.
-     */
-    onEscapeKeyUp: PropTypes.func, // eslint-disable-line react/sort-prop-types
-    /**
-     * Callback fired before the dialog exits.
-     */
-    onExit: PropTypes.func,
-    /**
-     * Callback fired when the dialog is exiting.
-     */
-    onExiting: PropTypes.func,
-    /**
-     * Callback fired when the dialog has exited.
-     */
-    onExited: PropTypes.func, // eslint-disable-line react/sort-prop-types
-    /**
-     * Callback fired when the dialog requests to be closed.
-     */
-    onRequestClose: PropTypes.func,
-    /**
-     * If `true`, the Dialog is open.
-     */
-    open: PropTypes.bool,
-    /**
-     * The CSS class name of the paper inner element.
-     */
-    paperClassName: PropTypes.string,
-    /**
-     * Transition component.
-     */
-    transition: PropTypes.oneOfType([PropTypes.func, PropTypes.element]),
-  };
+class Dialog extends PureComponent<DefaultProps, Props, void> {
+  props: Props;
 
-  static defaultProps = {
-    fullScreen: false,
+  static defaultProps: DefaultProps = {
+    fullScreen: 'sm',
```

Comment:
What do you think of the UI? I'm not sure it's in the specification.
![mai-19-2017 09-15-29](https://cloud.githubusercontent.com/assets/3165635/26237042/c1893ed8-3c73-11e7-9260-877167d48c93.gif)